language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/ConfigTomlValueGenerationUtil.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.langserver.common.utils;
+
+import io.ballerina.compiler.api.symbols.ArrayTypeSymbol;
+import io.ballerina.compiler.api.symbols.RecordFieldSymbol;
+import io.ballerina.compiler.api.symbols.RecordTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.UnionTypeSymbol;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Util used to generate value entries for Config.toml keys.
+ *
+ * @since 2201.10.0
+ */
+public final class ConfigTomlValueGenerationUtil {
+
+    private ConfigTomlValueGenerationUtil() {
+    }
+
+    /**
+     * Get {@link TomlEntryValue} for a configurable {@link TypeSymbol}.
+     *
+     * @param type Type symbol of the configurable
+     * @param basicType union of basic types
+     * @param anydataOrJson union of anydata and json
+     * @param confName name of the configurable
+     * @return value string wrapped in {@link  TomlEntryValue}
+     */
+    public static TomlEntryValue getDefaultValueStr(TypeSymbol type, TypeSymbol basicType,
+                                            TypeSymbol anydataOrJson, String confName) {
+        switch (type.typeKind()) {
+            case BOOLEAN -> {
+                return new TomlEntryValue("false", true);
+            }
+            case BYTE, INT, INT_SIGNED8, INT_SIGNED32, INT_UNSIGNED8, INT_UNSIGNED16, INT_UNSIGNED32, INT_SIGNED16 -> {
+                return new TomlEntryValue("0", true);
+            }
+            case FLOAT, DECIMAL -> {
+                return new TomlEntryValue("0.0", true);
+            }
+            case STRING, XML, ANYDATA, JSON, SINGLETON -> {
+                return new TomlEntryValue("\"\"", true);
+            }
+            case TUPLE -> {
+                return new TomlEntryValue("[]", true);
+            }
+            case ARRAY -> {
+                ArrayTypeSymbol arrayTypeSymbol = (ArrayTypeSymbol) type;
+                TypeSymbol memberType = CommonUtil.getRawType(arrayTypeSymbol.memberTypeDescriptor());
+                if (memberType.typeKind() == TypeDescKind.UNION) {
+                    memberType = ((UnionTypeSymbol) memberType).memberTypeDescriptors().get(0);
```

Comment:
Why we only consider the 1st value of the member-type descriptors?

Reply:
When generating a default value for a union type we should always consider the left most type among the member types.